### PR TITLE
[glm] Modernization

### PR DESCRIPTION
**Pull request by xiaozhuai**
1. Modernization
2. Add license

### CODE REVIEW

File: ports/glm/CMakeLists.txt [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
-    NAMESPACE glm::
-    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/glm
-)
-
-install(
-    DIRECTORY ${CMAKE_SOURCE_DIR}/glm
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-    FILES_MATCHING PATTERN "*.hpp*" PATTERN "*.inl*" PATTERN "*.h*"
-)
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
```
Comment (JonLiu1993) [inline ports/glm/CMakeLists.txt:1]: @xiaozhuai, Thanks for your PR, could you please keep the original format of EOL for this patch, maybe LF?

Comment (xiaozhuai) [reply ports/glm/CMakeLists.txt:1]: Original format mixing CRLF and LF, So I change it to LF.

**Comment by JonLiu1993** [comment]
**The usage test failed, the details are as follows:**

1)Configure CMake project:
```
1> [CMake] CMake Error at F:/Feature-test/glm/vcpkg/scripts/buildsystems/vcpkg.cmake:843 (_find_package):
1> [CMake]   Could not find a package configuration file provided by "glm" with any of
1> [CMake]   the following names:
1> [CMake] 
1> [CMake]     glmConfig.cmake
1> [CMake]     glm-config.cmake

```

<details>
<summary><b>CMakeProject42.cpp</b></summary>
<pre>


#include &#60;iostream&#62;
#include "glm/common.hpp"

int main()
{
	cout << "Hello CMake." << endl;
	return 0;
}

</pre></details>


<details>
<summary><b>CMakeLists.txt</b></summary>
<pre>
cmake_minimum_required (VERSION 3.8)

project(test)

find_package(glm CONFIG REQUIRED)
add_executable (main "CMakeProject42.cpp" "CMakeProject42.h")
target_link_libraries(main PRIVATE glm::glm)

**Comment by xiaozhuai** [comment]
> The usage test failed, the details are as follows:

This is odd, is ok on my machine. 

```cmake
cmake_minimum_required(VERSION 3.24)
project(test_glm)

set(CMAKE_CXX_STANDARD 14)

find_package(glm CONFIG REQUIRED)

add_executable(test_glm main.cpp)

target_link_libraries(test_glm PRIVATE glm::glm)
```

```cpp
#include <iostream>
#include <glm/common.hpp>

int main() {
    std::cout << "Hello, World!" << std::endl;
    return 0;
}
```

**Comment by xiaozhuai** [comment]
> ```
> 1> [CMake] CMake Error at F:/Feature-test/glm/vcpkg/scripts/buildsystems/vcpkg.cmake:843 (_find_package):
> 1> [CMake]   Could not find a package configuration file provided by "glm" with any of
> 1> [CMake]   the following names:
> 1> [CMake] 
> 1> [CMake]     glmConfig.cmake
> 1> [CMake]     glm-config.cmake
> ```

Maybe your CMAKE_TOOLCHAIN_FILE is not set correctly?

**Comment by xiaozhuai** [comment]
glm has some downstream port, If this error occured, CI wouldn't pass.

**Comment by JonLiu1993** [comment]
I deleted the cache and re-authenticated it successfully.
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\ENTERPRISE\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 16 2019" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\CMakeProject42\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/glm/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\CMakeProject42" 2>&1"
1> Working directory: C:\Users\test\source\repos\CMakeProject42\out\build\x64-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.29.30146.0
1> [CMake] -- The CXX compiler identification is MSVC 19.29.30146.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] -- Configuring done
1> [CMake] -- Generating done
1> [CMake] -- Build files have been written to: C:/Users/test/source/repos/CMakeProject42/out/build/x64-Debug
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted toolchain configurations.
1> Extracted includes paths.
1> CMake generation finished.
````

**Comment by BillyONeal** [comment]
Thanks for the modernization!

**Pull request merged**